code/handlers/permissions.q

```diff
@@ -189,8 +189,8 @@ mainexpr:{[u;e;b;pr]
 expr:mainexpr[;;runmode;permissivemode]
 allowed:mainexpr[;;0b;0b]
 
-parsequery:{[q]q:$[10=type q;q;10h=abs type f:first q;destringf[f],1_ q;q]}
-destringf:{$[(x:`$x)in key`.q;.q x;x~`insert;insert;x]}
+parsequery:{[q]q:$[10=type q;q;10h=abs type f:first q;destringf[f],1_ q;q]};
+destringf:{$[(s:`$x)in key`.q;.q s;s~`insert;insert;100h=type f: @[value; x; 0];f;s]};
```

Comment:
I think if we use `value` here, it could mean that we excute code before we've actually checked permissions? e.g. if first element of query is a string expression that _isn't_ a lambda, such as this query:

```q
(".jm.x:123";`)
```

Regardless of later permissions checks etc., the `.jm.x:123` expression would already have been executed here. I think `parse` should work here without potential side effects?

```suggestion
destringf:{$[(s:`$x)in key`.q;.q s;s~`insert;insert;100h=type f: @[parse; x; 0];f;s]};
```